### ISSUE TITLE
Subtract race canceled registrations from event/race participant counts

### ISSUE THREAD

**Issue by runbeyond** [spectrumtrail/spectrum#158]
As a business administrator, I want up to date participant counts per event/race at all times so that participant counts are accurate for planning purposes and so that relationship between participant counts and participant caps is always working harmoniously.

Acceptance criteria:
1) When a business manager cancels a race registration, the canceled registration is deducted from the race's participant count.
2) If a race is sold out and a registration is canceled, the race is automatically updated to reflect the fact that there is now registration availability.
3) Once canceled registrations are deducted from the participant count, the change is reflected in the participants section of the main admin dashboard.

**Comment by willblake01** [comment]
@runbeyond @ColinW520 I did a preliminary check into this and it looks like the issue might be with the race is_full? logic. Currently it's checking the participants with a payment against the participants cap to determine if a race is full and not subtracting cancellations. As a result, is_registerable? remains false even after cancelling a registration. 
![image](https://user-images.githubusercontent.com/16193879/93385830-61b93a80-f82c-11ea-8759-e25d40995aa6.png)
![image](https://user-images.githubusercontent.com/16193879/93386143-dee4af80-f82c-11ea-9674-b7863ee8b588.png)

**Comment by willblake01** [comment]
I need to keep looking into it but it looks like the start list is up to date because that logic takes cancellations into account. On first take when I use similar logic I seem to get the desired behavior.
![image](https://user-images.githubusercontent.com/16193879/93386544-7813c600-f82d-11ea-9bab-814fcdf1568c.png)
![image](https://user-images.githubusercontent.com/16193879/93386603-8eba1d00-f82d-11ea-8ec9-5f74af0f497c.png)

**Comment by runbeyond** [comment]
Nice work, Will! Let me know if I can do anything to help.

**Comment by ColinW520** [comment]
I believe this was desired behavior. Imagine if a race is full, and then mallory/jason add more people via bumping the limit. Then, some people cancel, but, mallory and Jason don't want to let any additional people in?

May not have a bug here. Just need to know: enforce the _live_ count -- or, enforce the 'max it has ever been' count.

**Comment by ColinW520** [comment]
@runbeyond ^^ need an answer here

**Comment by willblake01** [comment]
@runbeyond @ColinW520 Ok, I think I get what needs to be done now. Sounds like we want the the race model untouched so the user facing behavior stays the same but the view in the admin console needs to be updated to take into account the cancellations. Sound right???
![image](https://user-images.githubusercontent.com/16193879/93808644-e1bb1800-fc11-11ea-9508-c52a9379d80d.png)

**Comment by runbeyond** [comment]
That sounds right.

**Comment by runbeyond** [comment]
I would think we enforce live count. If we cancel a registration and don’t
want it to be filled, we can manually reduce the participant cap. I think
this how it works now and is what we do now.

In the admin dashboard, the participant count doesn’t update to reflect
cancellations, so when we make several cancellations like we had to do with
Sky Island this year, we no longer know how many people are in the races.

**Comment by runbeyond** [comment]
@ColinW520 and @willblake01 - I feel somewhat lost on this issue right now and I am confident it is my fault for not clearly articulating the problem statement upfront. I am going to try to do a better job here. Here is what I just did and what I would expect to happen...followed by a refreshed problem statement with an example from Sky Island 2020.

1) I registered for the Wonderland Marathon. 
Expected behavior: After I registered, the paid participant count for the Wonderland Marathon on the main admin dashboard changed from 8 paid participants to 9 paid participants. 
Actual behavior: After I registered, the paid participant count for the Wonderland Marathon on the main admin dashboard changed from 8 paid participants to 9 paid participants.

2) I canceled my Wonderland Marathon registration.
Expected Behavior: The paid participant count changes from 9 to 8 to reflect the canceled race registration
Actual behavior: The paid participant count did not change.

As I understand, the 'Spots Remaining' Column on the Main Admin Dashboard is calculated by subtracting 'Paid Participants' from 'Participant Cap'.

We're presented with two problems related to the behavior specified above:
1) The 'Paid Participant' count is inaccurate as soon as we cancel one registration. And it is unusable once we can cancel many registrations. 
2) Because canceled registrations are not factored out of the 'Paid Participants' count, the 'Spots Remaining' count is also not accurate.

Here is an example of the impact of these problems and how we have to work around them. Let's take the 2020 Sky Island Trail Race. Let's first assume the race has sold out. Then we had to cancel 100 registrations (for ease of explanation).

Let's say each race had sold out with 125 racers. And let's say we canceled 50 registrations in each race. Now that we have canceled 100 registrations, we want to allow the races to fill again. In order to allow more registrations, we have to raise the 'Participant Cap' for each race by 50 because the 'Paid Participants' count doesn't update and thus the 'Spots Remaining' count doesn't update.

Additionally, if we want to know how many participants are in each Sky Island race at any given time, we cannot know by looking at the 'Paid Participant' count because it doesn't update based on canceled registrations. So, we have to either count the start list on the event page or download the start list from the Admin Dashboard and count the participants.

I hope this makes sense. Are we all starting to get on the same page here?
